lektor/admin/static/js/components/DialogSlot.tsx

```diff
@@ -21,14 +28,14 @@ class DialogSlot extends React.Component {
     hub.unsubscribe(DialogChangedEvent, this.onDialogChanged);
   }
 
-  onDialogChanged(event) {
+  onDialogChanged(event: DialogChangedEvent) {
     this.setState({
       currentDialog: event.dialog,
       currentDialogOptions: event.dialogOptions || {},
     });
   }
 
-  initDialogInstance(dialog) {
+  initDialogInstance(dialog: DialogInstance | null) {
```

Comment:
Could this be `dialog?: DialogInstance` or doesn't it work that way?

My question apply to most changes where `T | null` is used. Since you use `?` in some places assume this is correct, but I haven't used TS enough to know the difference.

Reply:
It's not quite the same - `dialog?:` would turn this into an optional argument which could be omitted or set to `undefined`. Here I think that `null` fits better since it's a bit more explicit for a "missing" or "empty" value.